### PR TITLE
Issue 48 - Use high resolution assets when possible

### PR DESCRIPTION
**Pull request by tmathern**
Fix https://github.com/hlxsites/franklin-assets-selector/issues/48

Allows higher resolution selections by allowing more mimetypes to be picked as the rendition to use than only PNG.
When a JPEG asset is picked, it is converted into a PNG blog using a canvas.

Currently only JPEG and PNG mimetypes are supported, but the list can be extended in the future as the support for image conversion improves, and once we test and validate that it works.

### CODE REVIEW

File: blocks/aem-asset-selector/aem-asset-selector-util.js [hunk 1]
```diff
@@ -63,6 +63,13 @@ const REL_RENDITIONS = 'http://ns.adobe.com/adobecloud/rel/rendition';
 // TODO: change this to Asset Link IMS client ID
 const IMS_CLIENT_ID = 'p66302-franklin';
 const ASSET_SELECTOR_ID = 'asset-selector';
+const CLIPBOARD_SUPPORTED_BINARY_MIMETYPES = [
```
Comment (tmathern) [inline blocks/aem-asset-selector/aem-asset-selector-util.js:66]: Mimetypes supported "natively" by the clipboard API (from the browser)

File: blocks/aem-asset-selector/aem-asset-selector-util.js [hunk 2]
```diff
@@ -63,6 +63,13 @@ const REL_RENDITIONS = 'http://ns.adobe.com/adobecloud/rel/rendition';
...
+  'image/png',
+];
+const SUPPORTED_RENDITIONS_FORMATS = [
```
Comment (tmathern) [inline blocks/aem-asset-selector/aem-asset-selector-util.js:69]: Renditions mimetypes that we can use in Franklin documents (needed for filtering).

File: blocks/aem-asset-selector/aem-asset-selector-util.js [hunk 3]
```diff
@@ -200,9 +207,9 @@ function getCopyRendition(asset) {
     // TODO: the clipboard API only supports using PNGs as the blob, so
     // only using PNG renditions. Will be fixed to allow altnernate
     // formats in a follow-up ticket
-    .filter((rendition) => rendition.type === 'image/png')
     .forEach((rendition) => {
-      if ((!maxRendition || maxRendition.width < rendition.width)) {
+      if (SUPPORTED_RENDITIONS_FORMATS.includes(rendition.type)
```
Comment (tmathern) [inline blocks/aem-asset-selector/aem-asset-selector-util.js:211]: Only look at mimetypes we can convert to use with the clipboard APIs...

File: blocks/aem-asset-selector/aem-asset-selector-util.js [hunk 4]
```diff
@@ -325,9 +375,9 @@ export async function copyAssetWithRapi(asset) {
       logMessage('Rendition download JSON not provided');
       return false;
     }
-    await copyToClipboardWithBinary(downloadJson.href, downloadJson.type);
+    await copyToClipboardWithBinary(downloadJson.href, downloadJson.type, asset);
```
Comment (tmathern) [inline blocks/aem-asset-selector/aem-asset-selector-util.js:378]: Asset info is needed to give the canvas the proper size

**Review by sdmcraft — APPROVED** [review]
Thanks! Couple of minor questions:
 I noticed a bit of latency in the image paste process in the recording. Is it because of the conversion? 
Also does it retain the image quality as such?

**Comment by tmathern** [comment]
> Thanks! Couple of minor questions: I noticed a bit of latency in the image paste process in the recording. Is it because of the conversion? Also does it retain the image quality as such?
> 
> Also please address linting errors and PSI failure before merging.

No, the latency is from the copy and all that happens. I had latency with copying the whole day, before and after those changes. (And loading too, but I managed to make a video in which loading didn't take 5 minutes).

**Pull request merged**